third_party/amd/include/Analysis/RangeAnalysis.h

```diff
@@ -124,6 +124,9 @@ bool cmpIIsStaticallyTrue(const DataFlowSolver &solver, arith::CmpIOp cmpOp);
 
 bool isEmptyInitializedRange(ConstantIntRanges rv);
 
+void populateFoldTrueCmpIOpPatterns(RewritePatternSet &patterns,
```

Comment:
Why this needs to be a `shared_ptr`? I think we populate the patterns in a pass, and the pass lifetime should be a strict superset of patterns? A raw pointer would do here?

Reply:
[createDataFlowSolver](https://github.com/triton-lang/triton/blob/74b4714d1a9f85df101822b44a06e8f92fe4dc90/include/triton/Analysis/Utility.h#L272) returns a smart pointer `unique_ptr` and I just made it shared because it's used in multiple places. I can change it to raw pointer.

Reply:
done